### PR TITLE
VACMS-16233 make script batches pick up where they left off

### PR DESCRIPTION
**Pull request by swirtSJW**
## Description

Relates to #16233

This makes it possible to resume the menu altering script from any given point if the script dies like it did the [first two attempts to run it on prod.](https://dsva.slack.com/archives/CT4GZBM8F/p1707842079725229?thread_ts=1707759434.620499&cid=CT4GZBM8F) 

This creates the ability for the execution of batched scripts to keep state using the Drupal State API  to keep track of the last successfully completed  batch and pick up where it left off on subsequent runs.

On prod the script ran out of memory processing  run number 118.   It is most likely that 118 ran successfully and it was actually 119 that failed, but to be safe, we will run it from 117 on by doing the following

- Set the last run state  to 117 with
  `ddev drush state:set script_library__va_gov_vamc_get_system_menus 117`
- Run the script with `drush scr ./scripts/content/VACMS-16233-reorder-vamc-menu-items.php`

## QA steps

- [x] Login to [Tugboat dashboard](https://tugboat.vfs.va.gov/65cc20c5067b1809aa527bf6) 
- [x] Open the php terminal
- [x] run `ddev drush state:set script_library__va_gov_vamc_get_system_menus 117`
- [x] run `drush scr ./scripts/content/VACMS-16233-reorder-vamc-menu-items.php`
- [x] Validate output looks like: 
image
- [x] When complete, validate that VA Spokane health care has been re-ordered (the first item in the resumed
image

- [x] validate that VA St Cloud menu has been rearranged (to rule out off-by-one error)
image

- [x] validate VA Lebanon has been  rearranged (to rule out off by one error on other end - the last one in the list)
image

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.
- [ ] If there are field changes, front end output has been thoroughly checked.

### Select Team for PR review

- [ ] `CMS Team`
- [ ] `Public websites`
- [ ] `Facilities`
- [ ] `User support`
- [ ] `Accelerated Publishing`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing announcement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping the UX writer so they are ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping the UX writer to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

**Review by omahane — COMMENTED** [review]
This looks great. Getting and setting state for a series of script runs should be useful going forward, too.

**Review by omahane — APPROVED** [review]
I see no issues here.

File: scripts/content/script-library.php [hunk 1]
```diff
@@ -341,17 +353,23 @@ function script_library_sandbox_init(array &$sandbox, $counter_callback, array $
...
     '@completed' => $sandbox['current'],
+    '@element' => $sandbox['element'],
     '@total' => $sandbox['total'],
   ];
-  $message = t('Processing... @completed/@total.', $vars) . PHP_EOL;
+
+  $message = t('Processed @element. @completed/@total.', $vars) . PHP_EOL;
   // Log the all finished notice.
   if ($sandbox['#finished'] === 1) {
     Drupal::logger('va_gov_vamc')->log(LogLevel::INFO, $completed_message, $vars);
```
Comment (swirtSJW) [inline scripts/content/script-library.php:368]:
```suggestion
    Drupal::logger('script_library')->log(LogLevel::INFO, $completed_message, $vars);
```

**Review by edmund-dunn — APPROVED** [review]
Nicely done! Approved!

**Pull request merged**